portal-frontend/src/app/services/application-submission/application-submission-draft.service.ts

```diff
@@ -0,0 +1,82 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { firstValueFrom } from 'rxjs';
+import { environment } from '../../../environments/environment';
+import { OverlaySpinnerService } from '../../shared/overlay-spinner/overlay-spinner.service';
+import { ToastService } from '../toast/toast.service';
+import {
+  ApplicationSubmissionDetailedDto,
+  ApplicationSubmissionDto,
+  ApplicationSubmissionUpdateDto,
+} from './application-submission.dto';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ApplicationSubmissionDraftService {
+  private serviceUrl = `${environment.apiUrl}/application-edit`;
+
+  constructor(
+    private httpClient: HttpClient,
+    private toastService: ToastService,
+    private overlayService: OverlaySpinnerService
+  ) {}
+
+  async getByFileId(fileId: string) {
+    try {
+      return await firstValueFrom(
+        this.httpClient.get<ApplicationSubmissionDetailedDto>(`${this.serviceUrl}/${fileId}`)
+      );
+    } catch (e) {
+      console.error(e);
+      this.toastService.showErrorToast('Failed to load Application, please try again later');
+      return undefined;
+    }
+  }
+
+  async updatePending(fileId: string, updateDto: ApplicationSubmissionUpdateDto) {
+    try {
+      this.overlayService.showSpinner();
+      const result = await firstValueFrom(
+        this.httpClient.put<ApplicationSubmissionDetailedDto>(`${this.serviceUrl}/${fileId}`, updateDto)
+      );
+      this.toastService.showSuccessToast('Application Saved');
+      return result;
+    } catch (e) {
+      console.error(e);
+      this.toastService.showErrorToast('Failed to update Application, please try again');
+    } finally {
+      this.overlayService.hideSpinner();
+    }
+
+    return undefined;
+  }
+
+  async publish(fileId: string) {
+    try {
+      this.overlayService.showSpinner();
+      // await firstValueFrom(
```

Comment:
is this still relevant?

Reply:
This is more placeholder code for the next ticket to implement publish